config/reader.go

```diff
@@ -331,14 +338,14 @@ type tlsConfig struct {
 func (tls *tlsConfig) readFromCache(id string, logger logging.Logger) error {
 	cachedCfg, err := readFromCache(id)
 	switch {
-	case os.IsNotExist(err):
+	case errors.Is(err, fs.ErrNotExist):
 		logger.Warn("No cached config, using cloud TLS config.")
 	case err != nil:
 		return err
 	case cachedCfg.Cloud == nil:
 		logger.Warn("Cached config is not a cloud config, using cloud TLS config.")
 	default:
-		if err := cachedCfg.Cloud.ValidateTLS("cloud"); err != nil {
+		if err := cachedCfg.Cloud.ValidateTLS("cloud"); err != nil && !cachedCfg.Cloud.SignalingInsecure {
```

Comment:
I could add this to `ValidateTLS`, but it seems weird to have `ValidateTLS` shortcut if it is invalid. Seems safer to be clear that we are ignoring the value of validateTLS in some circumstances

Reply:
[this is a matter of opinion and this is not my code-base, so if there is another way to write this that is clearer, I am totally open to suggestions]

Reply:
I think this is fine, but can you add a comment on why we are ignoring?

Reply:
Added comment and broke out if to be more clear in 140fd69e1

Reply:
great!